thruster_control/src/thruster_control.cpp

```diff
@@ -222,5 +229,25 @@ double ThrusterControl::convertRPMToRadSec(double rpms) { return (rpms * RPM_TO_
 
 double ThrusterControl::convertRadSecToRPM(double radsec) { return (radsec * RADSEC_TO_RPM); }
 
+void ThrusterControl::handle_rosmonFaults(const rosmon_msgs::State& msg)
+{
+  for (int i = 0; i < msg.nodes.size(); i++)
+  {
+    if (msg.nodes[i].state == rosmon_msgs::NodeState::CRASHED)
+    {
+      if (msg.nodes[i].name == "mission_manager_node" || msg.nodes[i].name == "health_monitor_node")
+      {
+        mission_manager::ReportExecuteMissionState abortMissionMsg;
+        abortMissionMsg.header.stamp = ros::Time::now();
+        abortMissionMsg.execute_mission_state = mission_manager::ReportExecuteMissionState::ABORTING;
+        publisher_MissionStateAbort.publish(abortMissionMsg);
```

Comment:
@patapufate this is awkward. Why should an actuator set a mission state?

Reply:
After meeting I'm closing this PR and work in a new one.